lib/src/connector/connection.cc

```diff
@@ -492,11 +493,22 @@ WS_Context_Ptr Connection::onTlsInit(WS_Connection_Handle hdl)
 void Connection::onClose(WS_Connection_Handle hdl)
 {
     Util::lock_guard<Util::mutex> the_lock { state_mutex_ };
-    timings.close = Util::chrono::high_resolution_clock::now();
+    timings.setClosed();
     auto con = endpoint_->get_con_from_hdl(hdl);
-    LOG_DEBUG("WebSocket on close event: {1} (code: {2}) - {3}",
-              con->get_ec().message(), con->get_remote_close_code(),
-              timings.toString());
+    auto close_code = con->get_remote_close_code();
+
+    if (close_code == 1000) {
+        // Normal closure; don't log error code
+        LOG_DEBUG("WebSocket on close event (nromal) - {1}", timings.toString());
+    } else {
+        LOG_DEBUG("WebSocket on close event: {1} (code: {2}) - {3}",
+                  con->get_ec().message(), close_code, timings.toString());
+    }
+
+    if (timings.isClosingStarted())
```

Comment:
Yes, only when the client starts the Closing Handshake with the synch `close()`.